docs/pktoken.md

```diff
@@ -0,0 +1,702 @@
+BASE64URL(UTF8(JWS Protected Header-COS)) || ':' ||
+BASE64URL(UTF8(JWS Signature-COS)) || '.' ||
+BASE64URL(JWS Refreshed Payload) || '.' ||
+BASE64URL(UTF8(JWS Refreshed Protected Header-OP)) || ',' ||
+BASE64URL(UTF8(JWS Refreshed Signature-OP)) || '.' ||
+```
+
+# Notes
+
+## GQ Signatures
+
+
+GQ (Guillou and Quisquater) signatures were invented in the paper ['A “Paradoxical” Indentity-Based Signature Scheme Resulting from Zero-Knowledge' (1988).](https://link.springer.com/content/pdf/10.1007/0-387-34799-2_16.pdf) GQ signatures were standardized in  GQ1 in [ISO/IEC 14888-2:2008](https://www.iso.org/standard/44227.html). Our GQ signatures are based on this standard but we have increased the security parameter to 256-bits.
+
+## Our JWS conventions
+
+In this section we discuss how the patterns we use for PK Tokens can be more broadly used for JSON Web Signatures in general.
+
+### The TYP pattern
+
+Because a PK Token always has more two or more signatures we have been forced to think about how to effective organize a JWS (JSON Web Signatures) that two or more signatures. If a JWS has only one signature, then the `iss` (issuer) claim in the payload identifies the party that generated the signature. Once you have two or more signatures, how do you determine which signature matches the issuer? How do you determine the identity of signer that generated each signature? Determining what party generated what signature is important because we need to know where to find which public key to verify which signature.
+
+One approach is to use signature order. For instance, we could specify that the first signature in the signatures list is the party that created the payload, the second signature is the cosigner, and so on. This approach has two major drawbacks. First, the order of the signatures is not signed or enforced in anyway. This means we can not assume that software and libraries won't reorder the signature list, breaking our ability to match signers to signatures. Second, this approach doesn't solve the problem of optional signers, who may or may not be required.
+
+Instead we use the `typ` (type) key in the protected header to specify the different types of signatures. For instance if the protected header of a signature has `typ=COS` it is a cosigner signature. In JWT's it is customary to have a `typ=JWT`. We extend this so that signatures each signing party in a protocol specifies their role in the `typ` key of their protected header.
+
+### Protected header claims
+
+In a JWT the claims the issuer is making are set in the payload. However this isn't sufficient for OpenPubkey, parties may wish to add additional claims although with their signature. For instance a signing party who independently authenticated the identity, might want to add an additional claim specifying the time at which this independent authentication took place. This signature can not update the payload without breaking the signature that already signed the payload.
```

Comment:
I think you meant to say "along with their signature" instead of "although"

Reply:
fixed